### ISSUE TITLE
Firefox 46.0.1 requires MPROTECT disabled for plugin container

### ISSUE THREAD

**Issue by fowlslegs** [freedomofpress/securedrop#1307]
As it is, all functional tests running FF 46.0.1 will fail under a grsec kernel. This makes basic testing in our app-staging and development VMs impossible. The solution for me was to disable MPROTECT on the FF plugin container with the following commands:

```
setfattr -n user.pax.flags -vm /usr/lib/firefox/firefox
setfattr -n user.pax.flags -vm /usr/lib/firefox/plugin-container
```

This should probably be done automatically from now on as a precaution. Obviously there no need to worry about security ramnifications (even accidental) as this code belongs under the app-test role.

**Comment by fowlslegs** [comment]
See https://github.com/freedomofpress/securedrop/issues/1301#issuecomment-220525681

**Comment by conorsch** [comment]
Great catch. We've been running the app tests on the `development` VM, but staging pulls in the same dependencies to afford testing. As far as the solution goes, near-term we should probably use `paxctl` rather than `setfattr` because that's what we're doing elsewhere. Further down the road we can consider using [paxctld](https://github.com/freedomofpress/ansible-role-paxctld), which uses extended attributes like `setfattr`.

**Comment by garrettr** [comment]
This is a small enough change with a big enough benefit (makes it easy to run the test suite in the staging environment for 0.3.7) that I think we should add some automation for this.

**Comment by conorsch** [comment]
Resolved by #1310 for next release. Backporting to develop tracked in #1313.
